src/jvmMain/kotlin/com/charleskorn/kaml/YamlOutput.kt

```diff
@@ -77,14 +77,17 @@ internal class YamlOutput(
             currentTypeName = value
             shouldReadTypeName = false
         } else {
-            emitQuotedScalar(value)
+            when {
+                value.contains('\n') -> emitQuotedScalar(value, configuration.multiLineStringStyle.scalarStyle)
+                else -> emitQuotedScalar(value, configuration.quotedScalarStyle.scalarStyle)
+            }
         }
     }
 
-    override fun encodeEnum(enumDescriptor: SerialDescriptor, index: Int) = emitQuotedScalar(enumDescriptor.getElementName(index))
+    override fun encodeEnum(enumDescriptor: SerialDescriptor, index: Int) = emitQuotedScalar(enumDescriptor.getElementName(index), configuration.quotedScalarStyle.scalarStyle)
 
     private fun emitPlainScalar(value: String) = emitScalar(value, ScalarStyle.PLAIN)
-    private fun emitQuotedScalar(value: String) = emitScalar(value, ScalarStyle.DOUBLE_QUOTED)
+    private fun emitQuotedScalar(value: String, scalarStyle: ScalarStyle = ScalarStyle.DOUBLE_QUOTED) = emitScalar(value, scalarStyle)
```

Comment:
Is the default value needed?

```suggestion
    private fun emitQuotedScalar(value: String, scalarStyle: ScalarStyle) = emitScalar(value, scalarStyle)
```

Reply:
it's used below for the emitting the type tag, I can remove the default value here and move it down there, or should that honour the `quotedScalarStyle` also?

Reply:
Ah, I'd missed that usage. I'd remove the default value and make it explicit in the caller.